### PR TITLE
[Feature] Comprehensive Method-Level Tracing with IL Weaving

### PR DESCRIPTION
**Pull request by Charles-aelf**
### Overview
This PR aims to reduce MTTR by implementing a comprehensive method-level tracing solution using IL weaving. It provides a generic way for the team to enable method level trace / exception analysis without writing any customised code.
### Key Features
- IL Weaving: Complete compile-time IL weaving which does not incur any runtime performance degradation
- Method-Level Tracing: Comprehensive tracing covering method entry, exit, and exception handling
- External Trace ID Support: Runtime configuration of external trace IDs via middleware and agent RequestContext
- Multi-Threading Support: Robust trace ID propagation across application domains and concurrent requests
- Performance Optimised: Benchmarking suite with performance comparison metrics
### Technical Changes
- Framework Restructure: Moved tracing components to appropriate projects and modules
- Silo Integration: Integrated tracing configuration into Orleans silo setup
- Admin API: Added tracing controller to admin API module with filtering capabilities
- Test Coverage: Comprehensive E2E and unit tests covering middleware, concurrency, and edge cases
### Documentation & Benchmarks
- Added comprehensive architecture documentation
- Performance benchmarking suite with detailed metrics
- Sample projects demonstrating IL weaving usage

### E2E tests

- Add Trace Id into Track Id List

Screenshot 2025-08-26 at 15 46 32

- Trigger the API Request
Screenshot 2025-08-26 at 15 46 32

- Observe the trace log at Jaegar

Screenshot 2025-08-26 at 15 46 08

- Remove Trace Id from Track Id List
Screenshot 2025-08-26 at 17 54 46

- Fire API Request again. No trace entry observed
Screenshot 2025-08-26 at 17 57 29

### CODE REVIEW

File: framework/samples/FodyTracingWeaverDemo/MethodTracingWeaver.cs [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+
+        // Get trace attribute properties
+        var operationName = GetAttributePropertyValue<string>(traceAttribute, "OperationName", method.Name);
+        var captureParameters = GetAttributePropertyValue<bool>(traceAttribute, "CaptureParameters", true);
+        var captureReturnValue = GetAttributePropertyValue<bool>(traceAttribute, "CaptureReturnValue", true);
+        var maxCaptureSize = GetAttributePropertyValue<int>(traceAttribute, "MaxCaptureSize", 1000);
+
+        // Insert simple console logging for now - just get basic tracing working
+        processor.InsertBefore(firstInstruction, Instruction.Create(OpCodes.Ldstr, $"TRACE: Entering {operationName}"));
+        processor.InsertBefore(firstInstruction, Instruction.Create(OpCodes.Call, GetMethod(FindType("System.Console"), "WriteLine", _stringType)));
```
Comment (zhifenglee-aelf) [inline framework/samples/FodyTracingWeaverDemo/MethodTracingWeaver.cs:139]: Probably should be injecting `Microsoft.Extensions.Logging.ILogger` instead of `System.Console`.

**Review by zhifenglee-aelf — CHANGES_REQUESTED** [review]

**Review by zhifenglee-aelf — APPROVED** [review]
LGTM
